cluster/ci/config/prow/openshift/build/prow_component.yaml

```diff
@@ -0,0 +1,90 @@
+apiVersion: v1
+kind: Template
+parameters:
+- name: NAME
+  displayName: Name of the prow component to build
+  required: true
+- name: SOURCE_URL
+  displayName: Source URL
+  description: Git source URL to use for building the image
+  value: https://github.com/openshift/test-infra.git
+- name: SOURCE_REF
+  displayName: Source reference
+  description: Git source ref to use for building the image
+  value: master
+objects:
+- kind: ImageStream
+  apiVersion: v1
+  metadata:
+    name: "${NAME}"
+- kind: BuildConfig
+  apiVersion: v1
+  metadata:
+    name: "${NAME}-binaries"
+  spec:
+    successfulBuildsHistoryLimit: 1
+    failedBuildsHistoryLimit: 2
+    output:
+      to:
+        kind: ImageStreamTag
+        name: "${NAME}:binaries"
+    source:
+      git:
+        uri: "${SOURCE_URL}"
+        ref: "${SOURCE_REF}"
+      type: Git
+      dockerfile: |
+        FROM golang:1.9.2
+
+        COPY . /go/src/k8s.io/test-infra
+        RUN cd /go/src/k8s.io/test-infra && \
+            go get -u github.com/golang/dep/cmd/dep && \
+            dep ensure && \
+            cd prow && \
+            mkdir -p /static && \
+            cp -r cmd/deck/static/* /static && \
+            CGO_ENABLED=0 go install -a -installsuffix cgo -ldflags '-w' ./cmd/... && \
```

Comment:
Shouldn't we build just the thing we care about? Probably minimal runtime impact either way though

Reply:
Maybe a follow-up although I am not crazy about it.